src/materialized/tests/sql.rs

```diff
@@ -544,3 +546,99 @@ fn test_temporary_views() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+// This test attempts to observe a linearizability violation by creating a set of
+// sources which are constantly being appended to, then creating a materialized
+// view of each of their sizes, then repeatedly reading from some subset of
+// those materialized views. If any of the sizes ever decrease from one read to
+// the next, linearizability has been violated.
+//
+// The sources are based off of n CSVs, each named in<i>.csv, a source for each
+// one named s<i> which tails the CSV, and a materialized view for each named
+// v<i>.
+//
+// N.B. this test currently fails and is ignored. TODO(justin): fix it.
+// N.B. this test also fails more reliably in release mode.
+#[test]
+#[ignore]
+fn test_linearizable() -> Result<(), Box<dyn Error>> {
+    ore::test::init_logging();
+    let config = util::Config::default();
+    config.logical_compaction_window(Duration::from_secs(60));
+    let server = util::start_server(util::Config::default())?;
+    let mut client = server.connect(postgres::NoTls)?;
+
+    const NUM_FILES: usize = 5;
+    const NUM_READS: usize = 1000;
+    const NUM_WRITES: usize = 100000;
+
+    let temp_dir = tempfile::tempdir()?;
+
+    // For each source we want to create, we spawn a thread that is constantly appending to a CSV,
+    // which we tail for the source.
+    for i in 0..NUM_FILES {
+        let path = Path::join(temp_dir.path(), format!("in{}.csv", i));
+        thread::spawn({
+            let mut file = File::create(&path)?;
+            move || {
+                for _ in 0..NUM_WRITES {
```

Comment:
All the files will move in relative lockstep. Some random sleeps may cause some more interesting event interleavings.